app/src/main/java/com/codesoom/assignment/controller/TaskController.java

```diff
@@ -0,0 +1,95 @@
+package com.codesoom.assignment.controller;
+
+import com.codesoom.assignment.domain.Task;
+import com.codesoom.assignment.service.TaskService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+
+    TaskService taskService;
+
+    public TaskController(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @GetMapping("{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Task findOne(@PathVariable Long id) {
+        Task task = taskService.findOne(id);
+        if (task.isEmpty()) {
+            throw new ResponseStatusException(NOT_FOUND, "Unable to find resource");
+        }
+        return task;
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<Task> findAll() {
+        return taskService.findAll();
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Task create(@RequestBody Task task) {
+        if (task.hasNotTitle()) {
+            throw new ResponseStatusException(NOT_FOUND, "Unable to find resource");
+        }
+
+        return taskService.create(task);
+    }
+
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Task put(@PathVariable Long id, @RequestBody Task task) {
+        return update(id, task);
+    }
+
+    @PatchMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Task patch(@PathVariable Long id, @RequestBody Task task) {
+        return update(id, task);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void delete(@PathVariable Long id) {
+        if (isNotExist(id)) {
+            throw new ResponseStatusException(NOT_FOUND, "Unable to find resource");
+        } else {
+            taskService.delete(id);
+        }
+    }
+
+    /**
+     * PUT/PATCH는 현재 동일한 동작으로 처리하기 위해 로직 공통화
+     * @param id
+     * @param task
+     * @return
+     */
+    public Task update(Long id, Task task) {
+        if (isNotExist(id) || task.hasNotTitle()) {
+            throw new ResponseStatusException(NOT_FOUND, "Unable to find resource");
+        }
```

Comment:
`isNotExist`는 해당하는 할 일을 찾고, 존재하지 않으면 false, 존재하면 true를 반환하는 것 같아요. 할 일이 존재하는지 힘들게 찾아왔는데, 등록할 할 일의 제목이 없어서 예외가 발생한다면, 찾은 사람 입장에서는 허무할 수 있겠죠. 실제 상황이었다면 그걸 왜 지금말해!! 라고 화를 낼수도 있을 것 같아요. 그래서 순서를 변경하는게 좋을 것 같습니다.

혹은 할 일을 조회하는 것과 등록할 할 일의 유효성을 검사하는 것이 다른 것이라고 생각하신다면 구분하여 작성할 수도 있을 것 같습니다.

```suggestion
if (task.hasNotTitle()) {
    throw new ResponseStatusException(BAD_REQUEST, "할 일이 이상해요. 할 일의 제목은 꼭 있어야 해요");
}

if (isNotExist(id)) {
    throw new ResponseStatusException(NOT_FOUND, "할 일을 찾을 수 없어요. 할 일의 아이디를 확인해 주세요");
}
```